src/mesh.jl

```diff
@@ -157,3 +205,134 @@ function _marchingmesh(ranges::NTuple{D,AbstractRange}, axes::SMatrix{D,D}) wher
+    (pt...) -> padright(pt, Val(L))
+
+#######################################################################
+# LinearMeshSpec
+#######################################################################
+struct LinearMeshSpec{N,L,T,R} <: MeshSpec{1}
+    vertices::SVector{N,SVector{L,T}}
+    samelength::Bool
+    closed::Bool
+    points::R
+end
+
+"""
+    linearmesh(nodes...; points = 13, samelength = false, closed = false)
+
+Create a `MeshSpec` for a one-dimensional `Mesh` connecting the `nodes` with straight
+segments, each containing a number `points` of points (endpoints included). If a different
+number of points for each of the `N` segments is required, use `points::NTuple{N,Int}`.
+If `samelength` each segment has equal length in mesh coordinates. If `closed` the last node
+is connected to the first node.
+
+# Examples
+
+```jldoctest
+julia> buildmesh(linearmesh(:Γ, :K, :M, :Γ; points = (101, 30, 30)))
+Mesh{1}: mesh of a 1-dimensional manifold
+  Vertices   : 159
+  Edges      : 158
+```
+
+# See also
+    `marchingmesh`, `buildmesh`
+"""
+linearmesh(nodes...; points = 13, samelength::Bool = false, closed::Bool = false) =
+    LinearMeshSpec(sanitize_BZpts(nodes), samelength, closed, points)
+
+function sanitize_BZpts(pts)
+    pts´ = parse_BZpoint.(pts)
+    dim = maximum(length.(pts´))
+    pts´´ = SVector(padright.(pts´, Val(dim)))
+    return pts´´
+end
+
+parse_BZpoint(p::SVector) = float.(p)
+parse_BZpoint(p::Tuple) = SVector(float.(p))
+function parse_BZpoint(p::Symbol)
+    pt = get(BZpoints, p, missing)
+    pt === missing && throw(ArgumentError("Unknown Brillouin zone point $p, use one of $(keys(BZpoints))"))
+    return SVector(float.(pt))
+end
+const BZpoints =
+    ( Γ  = (0,)
+    , X  = (pi,)
+    , Y  = (0, pi)
+    , Z  = (0, 0, pi)
+    , K  = (2pi/3, -2pi/3)
+    , Kp = (4pi/3, 2pi/3)
+    , M  = (pi, 0)
+    )
+
+linearmesh_nodes(l, br) = cumsum(SVector(0, segment_lengths(l, br)...))
+
+function segment_lengths(s::LinearMeshSpec{N,LS,TS}, br::SMatrix{E,LB,TB}) where {TS,TB,N,E,LS,LB}
+    T = promote_type(TS, TB)
+    verts = padright.(s.vertices, Val(LB))
+    dϕs = ntuple(i -> verts[i + 1] - verts[i], Val(N-1))
+    if s.samelength
+        ls = filltuple(T(1/(N-1)), Val(N-1))
+    else
+        ibr = pinverse(br)'
+        ls = (dϕ -> norm(ibr * dϕ)).(dϕs)
+        ls = ls ./ sum(ls)
+    end
+    return ls
+end
+
+function idx_to_node(s, br)
+    nodes = SVector.(linearmesh_nodes(s, br))
+    nmax = length(nodes)
+    nodefunc = nvec -> begin
+        n = only(nvec)
+        node = if n >= nmax
+            nodes[nmax]
+        else
+            nc = max(n, 1)
+            i = Int(floor(nc))
+            nodes[i] + rem(nc, 1) * (nodes[i+1] - nodes[i])
+        end
+        return node
+    end
+    return nodefunc
+end
+
+buildmesh(s::LinearMeshSpec{N,L,T}) where {N,L,T} = buildmesh(s, SMatrix{L,L,T}(I))
+
+function buildmesh(s::LinearMeshSpec{N}, br) where {N}
```

Comment:
With the previous definition of `points_per_segment`, `buildmesh` can be written as (such that the number of vertices in the returned mesh `==sum(resolution)`):
```suggestion
function buildmesh(s::Quantica.LinearMeshSpec{N}, br) where {N}
    
    nsegs = length(s.vertices)-1
    seg_npts = mypoints_per_segment(s, br)
    println(seg_npts)
    totalnpts = sum(seg_npts)
    nodes = Quantica.linearmesh_nodes(s, br)
    
    verts = [SVector(nodes[1])]
    for n in 1:nsegs
        for i in 1:seg_npts[n]
            push!(verts, SVector(nodes[n] + (nodes[n+1]-nodes[n])*i/seg_npts[n]))
        end
    end
    
    nv = length(verts)
    
    adjmat = sparse(vcat(1:nv-1, 2:nv), vcat(2:nv, 1:nv-1), true, nv, nv)
    s.closed && (adjmat[end, 1] = adjmat[1, end] = true)
    return Mesh(verts, adjmat)
end
```